intervals/main_objs.py

```diff
@@ -1751,191 +2077,389 @@ def presentationTypes(self, melodic_ngram_length=4, limit_to_entries=True,
                             for slist in list_combinations:
                                 temp = self._temp_dict_of_details(slist, entry_array, det, matches)
                                 temp["Presentation_Type"] = ImportedPiece._classify_by_offset(temp['Time_Entry_Intervals'])
-                                # print(temp)
                                 if 'PEN' in temp["Presentation_Type"]:
                                     points2 = points2.append(temp, ignore_index=True)
                                 if 'ID' in temp["Presentation_Type"]:
                                     points2 = points2.append(temp, ignore_index=True)
 
-            # points2 = points2[points2["Presentation_Type"].isin(hidden_types_list)]
-            # return(points2)
             points_combined = points.append(points2, ignore_index=True)
             points_combined["Offsets_Key"] = points_combined["Offsets"].apply(ImportedPiece._offset_joiner)
+            points_combined['Flexed_Entries'] = points_combined["Soggetti"].apply(len) > 1
+            points_combined["Number_Entries"] = points_combined["Offsets"].apply(len)
+            points_combined["Count_Offsets"] = points_combined["Offsets"].apply(set).apply(len)
+            points_combined = points_combined[points_combined["Count_Offsets"] > 1]
+            # points_combined = points_combined.sort_values("First_Offset").reset_index(drop=True)
+            points_combined = points_combined.reindex(columns=col_order).sort_values("First_Offset").reset_index(drop=True)
             points_combined.drop_duplicates(subset=["Offsets_Key"], keep='first', inplace=True)
+            # applying various private functions for overlapping entry tests.
+            # note that ng_durs must be passed to the first of these, via args
+            points_combined["Entry_Durs"] = points_combined[["Offsets", "Voices"]].apply(ImportedPiece._dur_ngram_helper, args=(ng_durs,), axis=1)
+            points_combined["Overlaps"] = points_combined[["Entry_Durs", "Offsets"]].apply(ImportedPiece._entry_overlap_helper, axis=1)
+            points_combined["Count_Non_Overlaps"] = points_combined["Overlaps"].apply(ImportedPiece._non_overlap_count)
+            points_combined.drop(['Count_Offsets', 'Offsets_Key', 'Entry_Durs', 'Overlaps'], axis=1, inplace=True)
```

Comment:
What happened to the memoization I put here?

Reply:
Sorry!